pontoon/sync/tests/test_tasks.py

```diff
@@ -259,6 +260,66 @@ def test_create_repository_log(self):
         log = RepositorySyncLog.objects.get(repository=repo.pk)
         assert_equal(log.repository, repo)
 
+    def test_no_pretranslation(self):
+        """
+        Ensure that pretranslation isn't called if pretranslation not enabled
+        or no new Entity, Locale or TranslatedResource is created.
+        """
+        self.mock_pull_changes.return_value = [True, {
+            self.repository.pk: Locale.objects.filter(pk=self.translated_locale.pk)
+        }]
+
+        sync_translations(self.db_project.pk, self.project_sync_log.pk,
+                          self.now, [], [], [], ['new_entity'])
+
+        # Pretranslation is not enabled
+        assert_false(self.mock_pretranslate.called)
+
+        self.db_project.pretranslation_enabled = True
+        self.db_project.save()
+
+        with self.patch('pontoon.sync.tasks.update_translated_resources', return_value=False):
+            sync_translations(self.db_project.pk, self.project_sync_log.pk, self.now)
+
+        # No new Entity, Locale or TranslatedResource
+        assert_false(self.mock_pretranslate.called)
+
+    def test_new_entities_pretranslation(self):
+        """
+        Test if pretranslation is called for newly added entities.
+        """
+        self.db_project.pretranslation_enabled = True
+        self.db_project.save()
+        self.mock_pull_changes.return_value = [True, {
+            self.repository.pk: Locale.objects.filter(pk=self.translated_locale.pk)
+        }]
+
+        with self.patch('pontoon.sync.tasks.update_translated_resources', return_value=False):
+            sync_translations(self.db_project.pk, self.project_sync_log.pk,
+                              self.now, [], [], [], ['new_entity'])
+
+        assert_true(self.mock_pretranslate.called)
+        assert_equal(self.mock_pretranslate.call_args[1]['entities'], ['new_entity'])
```

Comment:
Can you also ensure the `locales` argument of `pretranslate()` is accurately set?